packages/api-fetch/src/index.js

```diff
@@ -166,3 +167,35 @@ apiFetch.fetchAllMiddleware = fetchAllMiddleware;
 apiFetch.mediaUploadMiddleware = mediaUploadMiddleware;
 
 export default apiFetch;
+
+/**
+ * Function that fetches data using apiFetch, and updates the status.
+ *
+ * @param {string} path Query path.
+ */
+export function useApiFetch( path ) {
+	// Indicate the fetching status
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ data, setData ] = useState( [] );
+	const [ error, setError ] = useState( null );
+
+	useEffect( () => {
+		apiFetch( { path } )
```

Comment:
I might be wrong, but I'm wondering if the state should be reset prior to calling `apiFetch` by calling `setIsLoading`, `setData` and `setError` again with the initial values, as I think if the `path` changes, useState won't override any existing values being stored.

Reply:
Oooh good point, it probably won't because only the effect will re-run.